src/include/btmem.h

```diff
@@ -1303,7 +1303,7 @@ struct __wt_update {
      * from memory rather than using the local variable, mark the shared transaction IDs volatile to
      * prevent unexpected repeated/reordered reads.
      */
-    volatile uint64_t txnid; /* transaction ID */
+    wt_shared volatile uint64_t txnid; /* transaction ID */
 
     wt_timestamp_t durable_ts; /* timestamps */
```

Comment:
The timestamps should all be shared.

Reply:
This is a similar situation to the previous comment with possible WT_PUBLISH barriers required in [`__page_inmem_prepare_update`](https://github.com/wiredtiger/wiredtiger/blob/1befa2b41caad250425a84509c2464dd908f0ab6/src/btree/bt_page.c#L170). Happy to add them as needed.